_posts/2019-05-10-continuous-deployment-the-first-step-on-the-road-to-recovery.markdown

```diff
@@ -0,0 +1,63 @@
+
+One piece flow - a simple idea with a simple implementation: continuous deployment. Every commit flows through your continuous delivery pipeline and, assuming all conditions pass, makes its way to production.  The idea and the implementation are simple, but there are a number of fundamental principles that you must adhere to:
+ - You have a single, consistent pipeline flow of changes into production
+ - You must have confidence in your pipeline to publish changes to the next stage
+ - In order to prevent bottlenecks, you need a highliy automated pipeline
+ - In order to prevent bottlenecks, you need a high level of automated test coverage.
+ 
+Rather than making it an aspiration, we have found that starting with continuous deployment is the first step to maximising flow.  Fundamental to that approach is [trunk based development](https://trunkbaseddevelopment.com/).  Good practice, such as branch by abstraction and feature toggling occur as a result of the move to trunk based development in order to effectively manage complexity and maintain flexibility.  System architecture becomes simple, flexible and decoupled, driven by the need to reduce risk.
+
+In trunk based development, the team works from a single master branch, with short lived feature branches for managing review and CI cycles:
+[!Trunk Based Development diagram](../img/tbd.png)**
+
+If you have a dysfunctional delivery pipeline, continuous deployment can't happen overnight.  Automatically releasing to production would be folly.  Technical debt needs to be paid down in order to release the blockage.  This is waste that has always been there, but changing the process will make it visible. The following structured approach provide the metrics required to pay down the technical debt and remove the bottlenecks:
+1. Switch to Trunk Based development.  Tag releases to production, and use short lived release branches for hot-fixes 
+1. Map out your continuous delivery pipeline to production, including any manual steps.  
+1. Remove all manual approval gates up to production deployment. This immediately highlights waste in the value stream.    Issues found late in QA demonstrate a lack of automated testing; manual deployment processes increase lead times.  Complex deployment and extended outages highlight an inflexible architecture.
+1. Introduce one piece flow on master - if there are manual steps, this will highlight bottlenecks.
+
+We are consistently able to deploy changes, with confidence, to production multiple times per day.  We use stop-the-line processes to maintain flow.  We still have branches, and PRs, but continually look to improve by reducing their lifetime, including pairing.
+
+We have helped clients move from release cycles taking months, to days.
```

Comment:
A quote from a client or some kind of proof here would be amazing and really hit home with your article, not sure if we can get it though.